spread/google_test.go

```diff
@@ -0,0 +1,78 @@
+package spread_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/snapcore/spread/spread"
+
+	. "gopkg.in/check.v1"
+)
+
+type googleSuite struct{}
+
+var _ = Suite(&googleSuite{})
+
+func (s *googleSuite) TestPagination(c *C) {
+	n := 0
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch n {
+		case 0:
+			c.Check(r.ParseForm(), IsNil)
+			c.Check(r.Form.Get("pageToken"), Equals, "")
+			w.Write([]byte(`
+                {
+                    "items": [{"status":"READY","name":"ubuntu-1910-64-v20190826", "description":"ubuntu-19.10-64"}, {"status":"READY","name":"ubuntu-1904-64-v20190726", "description":"ubuntu-19.04-64"}],
+                    "nextpagetoken": "token-1"
+                }
+                `))
+		case 1:
+			c.Check(r.ParseForm(), IsNil)
+			c.Check(r.Form.Get("pageToken"), Equals, "token-1")
+			w.Write([]byte(`
+                {
+                    "items": [{"status":"READY","name":"opensuse-leap-42-3-v20190227", "description":"opensuse-leap-42-3-64"}, {"status":"READY","name":"debian-9-v20190901", "description":"debian-9-64"}],
+                    "nextpagetoken": "token-2"
+                }
+                `))
+		case 2:
+			c.Check(r.ParseForm(), IsNil)
+			c.Check(r.Form.Get("pageToken"), Equals, "token-2")
+			w.Write([]byte(`
+                {
+                    "items": [{"status":"READY","name":"fedora-28-64-v20181210", "description":"fedora-28-64"}],
+                    "nextpagetoken": ""
+                }
+                `))
+		}
+		n++
+	}))
+	defer mockServer.Close()
+
+	g := spread.NewGoogleProviderForTesting(mockServer.URL, nil, nil, nil)
+	c.Assert(g, NotNil)
+
+	images, err := g.ProjectImages("snapd")
+	c.Assert(err, IsNil)
+	c.Check(images, HasLen, 5)
+	i := 0
+	c.Check(images[i].Project, Equals, "snapd")
```

Comment:
nitpick: personally I think this would read better as a DeepEquals with the full images object compared, but this is okay too

Reply:
Totally agreed. This is a bit tricky because the struct is "googleImage" which is not exported. As this is a "spread_test" style test (i.e. no access to internals of the spread package) I couldn't do a DeepEqual because I can't do:
```
c.Check(images, DeepEquals, spread.googleImage{...})
```
Putting `type GoogleImage googleImage` into export_test.go also does not help. I could simply export the type but not sure if that is to Gustavos liking so I kept things in there as much as possible :)

Reply:
Ah right I didn't realize that, in that case this looks good, thanks for adding a comment about it